cmd/maestro/server/pulse_server.go

```diff
@@ -0,0 +1,120 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/openshift-online/maestro/pkg/api"
+	"github.com/openshift-online/maestro/pkg/client/cloudevents"
+	"github.com/openshift-online/maestro/pkg/dao"
+	"github.com/openshift-online/maestro/pkg/db"
+	"github.com/openshift-online/maestro/pkg/logger"
+)
+
+// PulseServer is a server that periodically updates the heartbeat for the current Maestro instance,
+// checks all Maestro instances' liveness and trigger statusresync for dead instances.
+type PulseServer struct {
+	instanceID    string
+	pulseInterval int64
+	checkInterval int64
+	instanceDao   dao.InstanceDao
+	lockFactory   db.LockFactory
+	sourceClient  cloudevents.SourceClient
+}
+
+func NewPulseServer() *PulseServer {
+	sessionFactory := env().Database.SessionFactory
+	return &PulseServer{
+		instanceID:    env().Config.MessageBroker.ClientID,
+		pulseInterval: env().Config.PulseServer.PulseInterval,
+		checkInterval: env().Config.PulseServer.CheckInterval,
+		instanceDao:   dao.NewInstanceDao(&sessionFactory),
+		lockFactory:   db.NewAdvisoryLockFactory(sessionFactory),
+		sourceClient:  env().Clients.CloudEventsSource,
+	}
+}
+
+// Start initializes and runs the pulse server.
+// It periodically updates the heartbeat for the current Maestro instance,
+// checks Maestro instances' liveness and trigger statusresync for dead instances.
+func (s *PulseServer) Start(ctx context.Context) {
+	logger := logger.NewOCMLogger(ctx)
+	instance := &api.ServerInstance{
+		Meta: api.Meta{
+			ID:        s.instanceID,
+			UpdatedAt: time.Now(),
+		},
+	}
+	_, err := s.instanceDao.UpSert(ctx, instance)
+	if err != nil {
+		logger.Error(fmt.Sprintf("Unable to create maestro instance: %s", err.Error()))
+		return
+	}
+
+	pulseTicker := time.NewTicker(time.Duration(s.pulseInterval) * time.Second)
+	checkTicker := time.NewTicker(time.Duration(s.checkInterval) * time.Second)
+	for {
+		select {
+		case <-ctx.Done():
+			pulseTicker.Stop()
+			checkTicker.Stop()
+			return
+		case <-pulseTicker.C:
+			logger.Infof("Updating heartbeat for maestro instance: %s", instance.ID)
```

Comment:
Can you add level for this logger? otherwise, it will fill up the screen with this logs.

Reply:
done.